src/Cache/Interceptors/AgentInterceptor.php

```diff
@@ -0,0 +1,23 @@
+<?php
+declare(strict_types=1);
+
+namespace Momento\Cache\Interceptors;
+
+use Grpc\Interceptor;
+
+class AgentInterceptor extends Interceptor
+{
+
+    private const AGENT = "php:0.3.0";
+    private bool $isFirstRequest = true;
+
+    public function interceptUnaryUnary($method, $argument, $deserialize, $continuation, array $metadata = [], array $options = [])
+    {
+        if ($this->isFirstRequest) {
+            $metadata["agent"] = [self::AGENT];
```

Comment:
If a user sets their own agent, this will silently overwrite it. Do we want that?

Reply:
The caller can't specify an agent. This is an internal value for metrics purposes.